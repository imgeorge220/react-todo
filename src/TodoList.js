import React, { useState } from 'react';
import Todo from './Todo';
import TodoForm from './TodoForm';
import uuid from "uuid/v4"
import { queryByTestId } from '@testing-library/react';

function TodoList() {
  const [todos, setTodos] = useState([]);

  const remove = id => {
    setTodos(todos.filter(todo => todo.id !== id));
  }

  const addTodo = todo => {
    let newTodo = { ...todo, id: uuid() };
    setTodos(oldTodos => [...oldTodos, newTodo]);
  }

  function editTodo(todo, id){
    const newTodos = [...todos]


    for(let i = 0; i < newTodos.length; i++){
      if(newTodos[i].id === id){
        newTodos[i] = todo;
      }
    }

    setTodos(newTodos);
  }
  

  return (
    <div className="TodoList">
      <TodoForm addTodo={addTodo} />
      <div data-testid="todo-container">
        {todos.map(todo =>
          <Todo
            key={todo.id}
            remove={remove}
            msg={todo.msg}
            id={todo.id}
            editTodo={editTodo}
          />
        )}
      </div>
    </div>
  );
}

export default TodoList;
