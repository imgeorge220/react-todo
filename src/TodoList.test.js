import React from "react";
import { render, queryByLabelText, fireEvent } from "@testing-library/react";
import TodoList from "./TodoList";


it("renders without crashing", function(){
  render(<TodoList />);
});

it("matches snapshot", function(){
  const { asFragment } = render(<TodoList />);
  expect(asFragment()).toMatchSnapshot();
});

it("shows a todo after submit", function (){
  const { getByLabelText, queryByText, queryByTestId } = render(<TodoList />);

  expect(queryByTestId("todo-container")).toBeInTheDocument();

  const todo = getByLabelText("Todo:")
  const submitBtn = queryByText("Add Todo!");

  fireEvent.change(todo, {target: {value: "new thing"}})
  fireEvent.click(submitBtn);

  expect(queryByTestId("todo")).toBeInTheDocument();
})

it("removes a todo after click", function(){
  const { getByLabelText, queryByText, queryByTestId } = render(<TodoList />);

  expect(queryByTestId("todo-container")).toBeInTheDocument();

  const todoInput = getByLabelText("Todo:")
  const submitBtn = queryByText("Add Todo!");

  fireEvent.change(todoInput, {target: {value: "new thing"}})
  fireEvent.click(submitBtn);

  let todo = queryByTestId("todo");

  expect(todo).toBeInTheDocument();

  fireEvent.click(todo);

  expect(todo).not.toBeInTheDocument();
})