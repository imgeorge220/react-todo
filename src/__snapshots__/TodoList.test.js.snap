// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="TodoList"
  >
    <div
      class="TodoForm"
    >
      <form>
        <label
          for="msg"
        >
          Todo:
        </label>
        <input
          id="msg"
          name="msg"
          value=""
        />
        <button
          type="submit"
        >
          Add Todo!
        </button>
      </form>
    </div>
    <div
      data-testid="todo-container"
    />
  </div>
</DocumentFragment>
`;
